calculator.rb

```diff
@@ -0,0 +1,174 @@
+	statement_first_num = statement.partition(/[\+\-\^\*\/\%]/)[0].strip
+	statement_operation = statement.partition(/[\+\-\^\*\/\%]/)[1].strip
+	statement_second_num = statement.partition(/[\+\-\^\*\/\%]/)[2].strip	
+
+	# run the calculate function and save the returned values into variables
+	value, operator = calculate(statement_operation, statement_first_num, statement_second_num)
+	
+	# return just the output value
+	return value
+end
+
+# Ask for what operation the user wants to perform and store the input in a variable.
+puts "What operation would you like to perform?"
+operation = gets.chomp.downcase
+
+# Create a boolean variable to store whether the operation is valid or not.
+valid = false
+
+# Check for operation validity
+while valid == false
+	# If operation falls into the flattened supported operations array, stop the loop and proceed.
+	if $supported_operations.flatten(2).include?(operation)
+		valid = true
+	# If the user pressed enter without entering data, ask them for another operation.
+	elsif operation == ""
+		puts "Please enter an operation."
+		operation = gets.chomp.downcase
+	# If the user entered a word that is not on the supported operations list, ask for another operation.
+	else
+		puts "I do not recognize the operation \"#{operation}\". Please enter: addition, subtraction, multiplication, division, exponent, or modulus."
+		operation = gets.chomp.downcase
+	end
+end
+
+# Ask the user for a number and store the input in a variable.
+puts "Enter your first number:"
+first_number_raw = gets.chomp
+
+# Reset the boolean variable to store the validity of the number
+valid = false
+
+# Run loop to check for number validity
+until valid == true
+	# Check to see if the first and last characters are (), then run evaluate_parens and feed the result into first number
+	if first_number_raw[0] == "(" && first_number_raw[-1] == ")"
+		first_number = evaluate_parens(first_number_raw).to_s
+		valid = true
+	# Check to make sure the input is a number
+	# If the input, when converted to a float then back to a string, or to an integer, then back to a string, does not retrain its identity
+	# or if the input is an empty string, then the number is invalid and a request is made for more input.
+	# Delete commas before processing to allow for numbers inputted with commas, like "2,000"
+	elsif first_number_raw.delete(",").to_f.to_s != first_number_raw.delete(",") \
+		&& first_number_raw.delete(",").to_i.to_s != first_number_raw.delete(",")
+		puts "Please enter a valid number."
+		first_number_raw = gets.chomp
+	# Otherwise the input is a valid number and the loop ends
+	else
+	 	first_number = first_number_raw
+		valid = true
+	end
+end
+
+# Perform the same operations to check for validity with the second number inputted.
+puts "Enter your second number:"
+second_number_raw = gets.chomp
+
+# Reset the validity variable
+valid = false
+
+# Run loop to check for number validity
+while valid == false
```

Comment:
It seems like you're repeating the same code from the first number input here. I think this would be another good candidate for changing into a function, so you only have to write the code once.